Merge Dev into master

Yttruire opened the pull request:
Essentially:
- Navigation bar (w/ navigation to home, settings, login/sign up/log out)
- Login/Signup popups
- General reorganisation of most user authentication components/folders
- Misc styling

Yttruire committed:
QOL UI Changes

Small changes to the UI to make it a little bit clearer, but to be further improved in the future

Yttruire committed:
Simple UI Changes

Some improvements to Landing Page UI
Password reset still work in progress and not available on frontend

Yttruire committed:
Updated LogIn and SignUp Component UI

Changed to use <Grid> from material-ui instead of <div> for consistency and easier styling in the future

Yttruire committed:
Added password reset email button

- Changed promps to alerts for error handling of firebase auth functions
- Fixed password reset button positioning
- Tested password reset button as working

Yttruire committed:
Installed Reactjs-popup

junwei26 committed:
removed firebase hosting set up successful message from webpage

junwei26 committed:
Added web app url to environment constants

junwei26 committed:
Added node to env

junwei26 committed:
Refactored web app and added protected routes

Yttruire committed:
Added Landing Page

- Redirected App to new landing page component (with app bar)
- Some UI changes to adjust for landing page

Yttruire committed:
Merge branch 'Dev' of https://github.com/junwei26/remiNdUS into Dev

Yttruire committed:
Merge landing page

- Duplicate landing pages have been merged
- All conflicts resolved
- Note that navigation to dashboard is not currently working
- Will fix where the display of the user information appears (Currently apepars when logged in and at landing page)

Yttruire committed:
Update Sign In/Sign Up Component

- UI Changes to spacing and some styling of the inputs

Yttruire committed:
(Not working) Setup for reorganisation of navigation bar

- Moved login and sign up pop up into its own component
- Added login and sign up button component
- Moved navigation bar into its own component
- Misc organisation changes

Yttruire committed:
(Not tested) Added UserAuthButton

- Wraps Login and Signup buttons

Yttruire committed:
Ran Prettier on all files

Prettier ran on all files

junwei26 committed:
Resolved routing issue during firebase hosting

Yttruire committed:
(Functional) UI Changes

- Flushed Login/Sign Up to the right
- Fixed Signup button display text

Yttruire committed:
Merge branch 'Dev' of https://github.com/junwei26/remiNdUS into Dev

Yttruire committed:
User Authentication UI Improvements

- Removed user auth wrapper
- Minor improvements to sign up pop-up

Yttruire committed:
Added Navigation Bar to Dashboard/Landing Page

Yttruire committed:
UI Improvement to Login Popup

Yttruire committed:
Final UI Improvement to Login/Signup button

- Changed popup size
- Styling of popup components

Yttruire committed:
Updated Navigation Bar with Home Button and Dashboard/Landing Page/Settings Page with Navigation Bar

- Installed material-ui icons
- Added account button on navigation bar
- Changed margin and padding of <body> to 0
- Added navigation from landing page -> dashboard -> settings
- Added Home button (to dashboard) as long as logged in
- Added navigation bar for all pages thus far

Yttruire committed:
Fixed REACT_APP_URL

Forgot to change back to our proper URL after testing oops

junwei26 commented on frontend/src/Components/pages/dashboard/index.js:
```diff
@@ -0,0 +1,18 @@
+  return (
+    <Grid container direction="column" justify="space-between" alignItems="stretch" spacing={2}>
+      <Grid item>
+        <NavigationBar />
```
We should probably shift the nav bar to the App.js level so it doesnt render a new one for each page

Yttruire replied:
This and the other two comments should be resolved in this commit https://github.com/junwei26/remiNdUS/pull/1/commits/fd0890f238843b571d600f8ca589655a002668e5
This is the first commit right after this PR.

junwei26 commented on frontend/src/Components/pages/landingpage/index.js:
```diff
@@ -0,0 +1,18 @@
+  return (
+    <Grid container direction="column" justify="space-between" alignItems="stretch" spacing={2}>
+      <Grid item>
+        <NavigationBar />
```
Same here

Yttruire replied:
Same as above reply

junwei26 commented on frontend/src/Components/pages/settingspage/index.js:
```diff
@@ -0,0 +1,61 @@
+      <>
+        <Grid container direction="column" justify="space-between" alignItems="stretch" spacing={2}>
+          <Grid item>
+            <NavigationBar />
```
Same here

Yttruire replied:
Same as above reply

Yttruire committed:
Shifted Navigation Bar to App.js

- Fixed entire navigation bar refreshing every page redirection
- (Issue) Navigation Bar appears to reset to the state when user isn't logged in when loading the page

Yttruire committed:
Added Password Reset form to Login Popup

- Moved Login form into its own component
- Moved password reset into its own form component
- Added password reset form to Login Popup

Yttruire committed:
Revert accidental change to .enc

Keep forgetting to change back after testing oop
